modules/services/syncthing.nix

```diff
@@ -78,10 +611,28 @@ in {
           };
 
           Service = {
+            ExecStartPre = mkIf (cfg.cert != null || cfg.key != null) "+${
+                pkgs.writers.writeBash "syncthing-copy-keys" ''
+                  syncthing_dir="''${XDG_STATE_HOME:-$HOME/.local/state}/syncthing"
+                  ${install} -dm700 "$syncthing_dir"
+                  ${optionalString (cfg.cert != null) ''
+                    ${install} -Dm400 ${
+                      toString cfg.cert
+                    } "$syncthing_dir/cert.pem"
+                  ''}
+                  ${optionalString (cfg.key != null) ''
+                    ${install} -Dm400 ${
+                      toString cfg.key
+                    } "$syncthing_dir/key.pem"
+                  ''}
+                ''
+              }";
             ExecStart = escapeShellArgs syncthingArgs;
             Restart = "on-failure";
             SuccessExitStatus = [ 3 4 ];
             RestartForceExitStatus = [ 3 4 ];
+            Environment =
+              mkIf (cfg.all_proxy != null) { inherit (cfg) all_proxy; };
```

Comment:
The original NixOS module adds `config.networking.proxy.envVars` to the environment, which seems to [contain not only `all_proxy` but also other, global proxy variables](https://github.com/NixOS/nixpkgs/blob/787623bca83b9d067a1eac881b24548b4003e5eb/nixos/modules/config/networking.nix#L210-L226), at least some of which (like `http_proxy` and `https_proxy`) are seemingly also [used by Syncthing](https://docs.syncthing.net/users/proxying.html). I'm not really familiar with proxy settings, but is there a reason not to add the remaining proxy variables like the original module does?

Reply:
In the `home-manager` module, I think `config` does not give us access to `networking.proxy.envVars` as it points to the `home-manager` configuration itself, and I am not quite sure how to access items in the root config. I would appreciate any ideas on how to do this.

Reply:
There is the `osConfig` module argument (see [Home Manager manual](https://nix-community.github.io/home-manager/index.xhtml#sec-install-nixos-module)). ~Although I'm not sure how this is handled if Home Manager is used standalone.~ In fact, some searching revealed that it's simply not available in a standalone installation.

I guess one could bind the module parameters via something like `@args{ config, lib, pkgs, ... }` and check for `osConfig` via `args ? osConfig` to include the system proxy variables if available.

Reply:
I force-pushed a version with `osConfig`, but a quick search doesn't appear to show any other modules using it :thinking:

Reply:
> I force-pushed a version with `osConfig`, but a quick search doesn't appear to show any other modules using it 🤔

I would assume that's because it's little known (I think the installation part of the manual is the only place where it is mentioned, and it's not the first place one would look for it) and since it works only if HM is used as a nixOS or nix-darwin module.

Personally I think it's ok to use as long as it's safe-guarded via something like `args ? osConfig`, to make sure the module does not break on standalone HM, but not sure how important the system proxy settings are.